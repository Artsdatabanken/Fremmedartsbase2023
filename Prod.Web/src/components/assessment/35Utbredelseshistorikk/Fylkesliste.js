import React from 'react'
import {observer} from 'mobx-react'
import * as Xcomp from '../observableComponents'

@observer
export default class Fylkesliste extends React.Component {
    render() {
        const style = {
            th: {},
            th2: {
                textAlign: "mid",
                display: "inline-block",
                WebkitWritingMode: "vertical-rl",
                MsMritingMode: "tb-rl",
                writingMode: "vertical-rl",
                xtransformOrigin: "50% 50%",
                xtransform: "rotate(-270deg)"
            }
        }
        const {columns, rows, countyLabel} = this.props
        return (
            <table>
                <tbody>
                    <tr>
                        <td>
                            <table className="table-striped">
                                <thead>
                                    <tr>
                                        <th>{countyLabel || "Fylke"}</th>
                                        {columns.map(column => <th key={column.title} style={style.th}>{column.title}</th>)}
                                    </tr>
                                </thead>
                                <tbody>
                                    {rows.map(row => <Fylke key={row.key} row={row} columns={columns}/>)}
                                </tbody>
                            </table>
                        </td>
                    </tr>
                </tbody>
            </table>
        )
    }
}

Fylkesliste.propTypes = {
    columns: React
        .PropTypes
        .arrayOf(React.PropTypes.shape({title: React.PropTypes.string, values: React.PropTypes.object.isRequired}))
        .isRequired,
    rows: React
        .PropTypes
        .arrayOf(React.PropTypes.shape({key: React.PropTypes.string, title: React.PropTypes.string}))
        .isRequired
}

const Fylke = ({row, columns}) => <tr>
    <td>{row.title}</td>
    {columns.map(column => <td key={column.title} style={{
        textAlign: "center"
    }}>
        <Xcomp.Bool observableValue={[column.values, row.key]}/>
    </td>)}
</tr>