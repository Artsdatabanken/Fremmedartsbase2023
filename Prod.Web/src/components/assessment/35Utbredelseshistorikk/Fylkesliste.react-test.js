import React from 'react'
import renderer from 'react-test-renderer'
import Fylkesliste from './Fylkesliste'

test('Fylkesliste', () => {
    const component = renderer.create(<Fylkesliste
        rows={[
        {
            key: 'OsA',
            title: 'Oslo'
        }, {
            key: 'He',
            title: 'Hedmark'
        }
    ]}
        columns={[
        {
            title: 'kjent',
            values: {
                'OsA': true
            }
        }, {
            title: 'antatt',
            values: {
                'He': true
            }
        }
    ]}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
