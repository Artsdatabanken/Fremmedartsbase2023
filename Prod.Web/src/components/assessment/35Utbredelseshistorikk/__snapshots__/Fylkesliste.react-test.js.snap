exports[`test Fylkesliste 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        <table
          className="table-striped">
          <thead>
            <tr>
              <th>
                Fylke
              </th>
              <th
                style={Object {}}>
                kjent
              </th>
              <th
                style={Object {}}>
                antatt
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Oslo
              </td>
              <td
                style={
                  Object {
                    "textAlign": "center",
                  }
                }>
                <input
                  checked={true}
                  disabled={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </td>
              <td
                style={
                  Object {
                    "textAlign": "center",
                  }
                }>
                <input
                  checked={false}
                  disabled={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </td>
            </tr>
            <tr>
              <td>
                Hedmark
              </td>
              <td
                style={
                  Object {
                    "textAlign": "center",
                  }
                }>
                <input
                  checked={false}
                  disabled={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </td>
              <td
                style={
                  Object {
                    "textAlign": "center",
                  }
                }>
                <input
                  checked={true}
                  disabled={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
